src/averageAges.js

```diff
@@ -37,9 +45,23 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womens = people.filter(person => person.sex === 'f');
+
+  if (withChildren === true) {
+    womens = womens.filter(woman => people.some(person =>
+      person.mother === woman.name));
+  }
+
+  const totalAge = womens.reduce((sum, person) => sum
+    + (person.died - person.born), 0);
+
+  const averageAge = parseFloat((totalAge / womens.length).toFixed(2));
+
+  return averageAge;
```

Comment:
Similar to the previous function, consider a more descriptive name than 'womens'. 
  The check 'withChildren === true' is redundant, you can directly use 'if (withChildren)'. 
  Also, consider a check for 'falsy' value instead of 'withChildren === false' for better readability.